components/datetime/src/provider/neo.rs

```diff
@@ -0,0 +1,155 @@
+use crate::pattern::{self, runtime};
+use icu_provider::prelude::*;
+
+use zerovec::{ule::UnvalidatedStr, VarZeroVec, ZeroMap};
+
+/// Symbols used for representing the year name
+///
+/// This uses an auxiliary key for length.
+#[icu_provider::data_struct(marker(YearSymbolsV1Marker, "datetime/symbols/years@1"))]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub enum YearSymbolsV1<'data> {
+    /// This calendar uses eras with numeric years, this stores the era names mapped from
+    /// era code to the name
+    Eras(#[cfg_attr(feature = "serde", serde(borrow))] ZeroMap<'data, UnvalidatedStr, str>),
```

Comment:
it occurs to me that all of the symbols keys are of the form "map OR vec" or "vec". There may be some value in having them all share the same "map or vec" type, reducing serde costs. This would mean some of the keys have discriminants they don't use, though.

Reply:
alternatively, I can make two structs: SymbolsLinearV1 and SymbolsLinearOrMapV1

(names up for debate)

I think that's the way to go.

Reply:
I went ahead and merged the linear ones